### ISSUE TITLE
IT timeout due to `ScaleResiliencyTest.ScaleDown`

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#15343]
**Summary**

> Try to answer the following as best as possible

- How often does the test fail? several times 
- Does it block your work? It stopped me from merging some changes
- Do we suspect that it is a real failure? 🤷 

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>GitHub Error</summary>
<pre>
The operation was canceled.
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

[io.camunda.zeebe.it.clustering.dynamic.ScaleResiliencyTest$ScaleDown-1-output.txt](https://github.com/camunda/zeebe/files/13451838/io.camunda.zeebe.it.clustering.dynamic.ScaleResiliencyTest.ScaleDown-1-output.txt)

**Additional**

I was able to find out which test was still running thanks to @deepthidevaki's awesome script:

```sh
grep "Running io." test.out > running.out

grep "Tests run:.*" test.out | grep "io." > testsRan.out
cut -d' ' -f3 running.out > testsRunning.out
cut -d'-' -f3 testsRan.out | cut -d' ' -f3 > testsRanClasses.out

grep -vf  testsRanClasses.out testsRunning.out 
```

**Comment by deepthidevaki** [comment]
Observations from the log:

Broker-2 has attempted to leave partition 2, but could not succeed because Broker-1 is not available. But it has already committed that first part of the joint consensus. When Broker-1 is restarted, in some cases partition 2 in Broker-2 is not yet started. In this case, Broker-0 which is the leader might commit the second part of joint consensus before Broker-2 is ready. When this happens, Broker-2 waits forever to become partition-2 to be ready. 

```
15:02:25.031 [Broker-0] [raft-server-2] [raft-server-0-2] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{raft-partition-partition-2}{role=CANDIDATE} - Cancelling election
15:02:25.032 [Broker-0] [raft-server-2] [raft-server-0-2] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-2} - Found leader 0
15:02:25.032 [Broker-0] [raft-server-2] [raft-server-0-2] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{raft-partition-partition-2} - Configuring 2
15:02:25.032 [Broker-0] [raft-server-2] [raft-server-0-2] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{raft-partition-partition-2} - Configuring 1
15:02:25.034 [Broker-2] [raft-server-2] [raft-server-2-2] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-2} - Found leader 0
```
Here leader is configuring both 1 and 2 with the first part of joint consensus which was already committed before the restart of broker-2.
```
15:02:25.035 [Broker-1] [raft-server-2] [raft-server-1-2] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Rejected InternalAppendRequest[term=3, leader=0, prevLogIndex=5, prevLogTerm=2, commitIndex=4, entries=[]]: Previous index (5) is greater than the local log's last index (3)
15:02:25.035 [Broker-0] [raft-server-2] [raft-server-0-2] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{raft-partition-partition-2} - Configuring 1
```
Here, Broker-0 configures Broker 1 with second part of joint consensus where Broker-2 is not part of the configuration anymore.
```
15:02:25.037 [Broker-1] [raft-server-2] [raft-server-1-2] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-2} - Setting firstCommitIndex to 4. RaftServer is ready only after it has committed events upto this index
15:02:25.037 [Broker-1] [raft-server-2] [raft-server-1-2] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-2} - Commit index is 4. RaftServer is ready
15:02:25.037 [Broker-1] [raft-server-2] [raft-server-1-2] DEBUG io.atomix.raft.impl.DefaultRaftServer - RaftServer{raft-partition-partition-2} - Server started successfully!
15:02:25.037 [Broker-1] [raft-server-2] [raft-server-1-2] INFO  io.atomix.raft.partition.impl.RaftPartitionServer - RaftPartitionServer{raft-partition-partition-2} - Server successfully bootstrapped partition PartitionId{id=2, group=raft-partition} in 4096ms
```
Broker-2 has already left the configuration, So it never becomes READY. As a result `startFuture` in Partition's StartupProcess is never completed.
```
15:02:32.091 [Broker-2] [Startup] [zb-actors-0] INFO  io.camunda.zeebe.broker.partitioning.PartitionManagerImpl - Leaving partition 2
15:02:32.094 [Broker-2] [Startup] [zb-actors-1] DEBUG io.camunda.zeebe.broker.partitioning.Partition - Shutdown was called with context: PartitionStartupContext{partition=2}
```
Leave completes successfully, as it has already left the configuration. So it tries to shutdown. But [shutdown is blocked because it is waiting on the `startupFuture`](https://github.com/camunda/zeebe/blob/b10310596bc38b644d062786efca505d8c51c8fd/scheduler/src/main/java/io/camunda/zeebe/scheduler/startup/StartupProcess.java#L198). 
```
15:04:20.968 [Broker-2] [Startup] [zb-actors-0] DEBUG io.camunda.zeebe.broker.partitioning.Partition - Shutdown was called with context: PartitionStartupContext{partition=2}
15:04:20.968 [Broker-2] [Startup] [zb-actors-0] INFO  io.camunda.zeebe.broker.partitioning.Partition - Shutdown already in progress
```
After the test fails, when it tries to shutdown the broker, we see that the shutdown of partition 2 is already in progress. Test is stuck waiting in the shutdown leading to the timeout of CI job.

The test passes when Broker-2 becomes ready before Broker-1 is up.
